### ISSUE TITLE
Sync DND

### ISSUE THREAD

**Issue by rkscodes** [sannidhyaroy/Soduto#5]
First of all, thank you very much for this fork. This is what i have been looking for.

Would it be possible to sync DND between mac and android ?

**Comment by sannidhyaroy** [comment]
This is a really convenient & I wanted to implement this. Surely it would be possible on the macOS side. However, I don't think the android app is capable of sending data packets to Soduto when `DND` is turned on/off on the android device. It would be very much possible to write a custom plugin for both KDE Connect and Soduto but you first need to ask the KDE Connect/Zorin Connect community to add this plugin or add it yourself. I'm pretty weak with Java, even though ironically that was my first programming language, so I'll only be able to manage the macOS side.

Here's the link to the plugins of KDE/Zorin Connect. You can read the `Readme.md` provided there to know how to add your own plugin:
[KDE Connect Plugins](https://github.com/KDE/kdeconnect-android/tree/master/src/org/kde/kdeconnect/Plugins)
[Zorin Connect Plugins](https://github.com/ZorinOS/zorin-connect-android/tree/master/src/org/kde/kdeconnect/Plugins)

### Basically, we need to add two plugins to both KDE/Zorin Connect as well as to Soduto:
1. To send a `data packet` of the `DND` status to the other devices, whenever it is turned on/off.
2. To set the `DND` status of the device, whenever it gets a data packet from another device that the `DND` status was changed.

**Comment by rkscodes** [comment]
Yeah sure, I will pitch this idea in the community you mentioned.
I think that for most people's workflow, imp thing is to control android's dnd from mac as we would do so while working or something which rarely happens on phone anyways. 
So I think bidirectional sync feature is nice to have but I think one-way sync would serve the purpose nonetheless.

**Comment by sannidhyaroy** [comment]
Yep, we can first implement syncing from Mac to Android & then vise-versa. Bidirectional sync is cool. Although, most people would actually use it to enable dnd on their phone when they want to focus because it's the phone pinging random notifications all the time.

**Comment by rkscodes** [comment]
Exactly, I agree.

**Comment by avichou** [comment]
it would be fantastic !!

**Comment by sannidhyaroy** [comment]
@avichou I agree. There are much more cool features that can be added but the main issue is that someone also needs to create a plugin for KDE/Zorin Connect to handle the android side of things.

**Comment by avichou** [comment]
> @avichou I agree. There are much more cool features that can be added but the main issue is that someone also needs to create a plugin for KDE/Zorin Connect to handle the android side of things.

True 

Honestly wondering how many people use kde connect , dunno how big the fanbase is tbh

**Comment by sannidhyaroy** [comment]
> Honestly wondering how many people use kde connect , dunno how big the fanbase is tbh

It pretty much consists of Linux users as a majority and the reason is obvious. Windows users rarely know about KDE Connect and now that we got Nearby Share on Windows too, the fanbase will even decrease among Windows Users. And macOS users with iPhones won’t research about it anyway to know. I have an android cuz I hate iPhones and we're a minority.

**Comment by rkscodes** [comment]
>   I have an android cuz I hate iPhones and we're a minority.
Exactly, I don't know how people settle with mac and iPhone, which feels on such opposite spectrum.

**Comment by sannidhyaroy** [comment]
> Exactly, I don't know how people settle with mac and iPhone, which feels on such opposite spectrum.

Ironically, my android works better alongside my mac. I used my dad's iPhone for a month just to experience how it feels with the ecosystem and although it's not bad, I feel my android is better. Airdrop works but takes some time with the "preparing" or "connecting" status or 10% of the time it straight doesn’t work, unlike Soduto which just instantly starts the transfer. My AirPods Pro work horribly when it comes to seamless switching, even though it was one of the main reasons I bought it. If you don’t know, the AirPods lineup have great audio quality but their ecosystem is only good in iPhones and iPads. macOS works like shit alongside AirPods.

AND THE MOST IMPORTANT! Charging!
I just carry my 50W powerbank with my devices and that's all. It charges all my devices. With an iPhone I have to carry what and what not. I hate lightning and how Apple gets a cut for every 3rd party lightning cable too for their MFI Program. It's just a scam~

**Comment by avichou** [comment]
> > Honestly wondering how many people use kde connect , dunno how big the fanbase is tbh
> 
> It pretty much consists of Linux users as a majority and the reason is obvious. Windows users rarely know about KDE Connect and now that we got Nearby Share on Windows too, the fanbase will even decrease among Windows Users. And macOS users with iPhones won’t research about it anyway to know. I have an android cuz I hate iPhones and we're a minority.

same, IOS is extremely limiting and annoying in many ways, whilst being super expensive 

but yeah it's kinda sad.i hope this project wont die cuz it's really the only good alternative we have to share files and clipboard. speaking of clipboard is there a workaround, be it with root or whatever, that would allow for phone's clipbaord to be shared automatically ?

**Comment by avichou** [comment]
> > Exactly, I don't know how people settle with mac and iPhone, which feels on such opposite spectrum.
> 
> Ironically, my android works better alongside my mac. I used my dad's iPhone for a month just to experience how it feels with the ecosystem and although it's not bad, I feel my android is better. Airdrop works but takes some time with the "preparing" or "connecting" status or 10% of the time it straight doesn’t work, unlike Soduto which just instantly starts the transfer. My AirPods Pro work horribly when it comes to seamless switching, even though it was one of the main reasons I bought it. If you don’t know, the AirPods lineup have great audio quality but their ecosystem is only good in iPhones and iPads. macOS works like shit alongside AirPods.
> 
> AND THE MOST IMPORTANT! Charging! I just carry my 50W powerbank with my devices and that's all. It charges all my devices. With an iPhone I have to carry what and what not. I hate lightning and how Apple gets a cut for every 3rd party lightning cable too for their MFI Program. It's just a scam~

they will bring usb-c next year, but it'll have an IC chip , making it proprietary if you want to transfer files quickly, or even charging fast (will probably be locked to 5w if you don't use apple's special usb-c cable for iphone. such bastard honestly) 

yeah i've been using an iphone and it's actually a nightmare in so many aspects lol. i wouldnt say android is much easier though, since sharing files without any connection can be done via airdrop ,easily so . and your clipboard gets shared automatically, and you also have messages (on android you have google messages but it wont work on macOS offline sadly) and apple notes 

btw sorry for this long thread but what note taking app do you use ? i really miss apple notes as i need a simple and reliable note taking app that works offline and syncs betweeen devices, and so far only apple notes seems to be able to do it . even google keep doesnt work offline on mac

**Comment by sannidhyaroy** [comment]
> speaking of clipboard is there a workaround, be it with root or whatever, that would allow for phone's clipbaord to be shared automatically ?

Yes, there's a Riru/Zygisk Module called "[Clipboard Whitelist](https://github.com/Kr328/Riru-ClipboardWhitelist)" and I've been using it since I started using Soduto. I recommend it if you're rooted. If you're not rooted, I'm not sure if it supports Shizuku.

If you've Magisk with Zygisk enabled, you can install the module and reboot. Then open the Clipboard app and checkmark the apps that you want to give clipboard access (like KDE Connect or Zorin Connect in this case).

**Comment by sannidhyaroy** [comment]
> btw sorry for this long thread but what note taking app do you use ? i really miss apple notes as i need a simple and reliable note taking app that works offline and syncs betweeen devices, and so far only apple notes seems to be able to do it . even google keep doesnt work offline on mac

I use Google Keep on my Android which is available on all platforms and Notability on my iPad that gets synced to my Dropbox, so I can view them with the Dropbox app on Android too~

**Comment by sannidhyaroy** [comment]
> but yeah it's kinda sad.i hope this project wont die cuz it's really the only good alternative we have to share files and clipboard.

There's actually a really good alternative I found called [NearDrop](https://github.com/grishka/NearDrop) which basically uses Android's built-in Nearby Share. Recently Google unveiled the Windows client of Nearby Share (beta) and I was wondering it'd be cool if I could built a macOS client of Nearby Share but thankfully I found this today lol. It's pretty buggy at this moment and only transfers files, photos and videos but it is faster than Soduto's transfer speeds.

It's not a complete package however and is only for transferring content, much like AirDrop.

**Comment by avichou** [comment]
> > but yeah it's kinda sad.i hope this project wont die cuz it's really the only good alternative we have to share files and clipboard.
> 
> There's actually a really good alternative I found called [NearDrop](https://github.com/grishka/NearDrop) which basically uses Android's built-in Nearby Share. Recently Google unveiled the Windows client of Nearby Share (beta) and I was wondering it'd be cool if I could built a macOS client of Nearby Share but thankfully I found this today lol. It's pretty buggy at this moment and only transfers files, photos and videos but it is faster than Soduto's transfer speeds.
> 
> It's not a complete package however and is only for transferring content, much like AirDrop.

thanks ! I heard about it and tried it but it's not nearly as good as soduto. besides soduto speeds are hella fast for me ?

1)can't share files from mac to android , only android to mac 
also can't share clipboard y'know

2) can only share if connected to the same wifi , which brings me to point 3

3)nearby share doesnt allow you to have hospot turned on on android. and macOS does not allow you to create a hospot . 

Thus, if you are outside and have no wifi , it is impossible to share files , at all

meanwhile with soduto , an easy workaround concists in creating a hospot on your phone and connecting the phone to it, which will allow you to share files and sync clipboard 

And this works even if you have no 4g !!

**Comment by sannidhyaroy** [comment]
> thanks ! I heard about it and tried it but it's not nearly as good as soduto. besides soduto speeds are hella fast for me ?

I stated that it's not a replacement for Soduto and it's extremely buggy but this is a very recent project so it's expected to not be stable or usable right now. I'm just happy that people are trying to close the gap between android and mac~

**Comment by avichou** [comment]
> > thanks ! I heard about it and tried it but it's not nearly as good as soduto. besides soduto speeds are hella fast for me ?
> 
> I stated that it's not a replacement for Soduto and it's extremely buggy but this is a very recent project so it's expected to not be stable or usable right now. I'm just happy that people are trying to close the gap between android and mac~

Yeah my bad i realized that but only after i answered lol
